grpc-protocol/src/main/java/com/linecorp/armeria/client/grpc/protocol/UnaryGrpcClient.java

```diff
@@ -126,21 +148,25 @@ private static void checkGrpcStatus(@Nullable String grpcStatus, HttpHeaders hea
     }
 
     private static final class GrpcFramingDecorator extends SimpleDecoratingHttpClient {
+        private final SerializationFormat serializationFormat;
 
-        private GrpcFramingDecorator(HttpClient delegate) {
+        private GrpcFramingDecorator(HttpClient delegate, SerializationFormat serializationFormat) {
             super(delegate);
+            // Validated in the UnaryGrpcClient ctor.
+            this.serializationFormat = serializationFormat;
         }
 
         @Override
         public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) {
+            final boolean isGrpcWebText = UnaryGrpcSerializationFormats.isGrpcWebText(serializationFormat);
```

Comment:
Could determine this at the constructor so we don't have to check on every request.